src/main/java/com/rabbitmq/client/RpcClient.java

```diff
@@ -392,9 +392,14 @@ public String getRoutingKey() {
     /**
      * Retrieve the correlation id.
      * @return the most recently used correlation id
+     * @deprecated Only works for {@link IncrementingCorrelationIdGenerator}
      */
     public int getCorrelationId() {
-        return _correlationId;
+        if (_correlationIdGenerator instanceof IncrementingCorrelationIdGenerator) {
+            return ((IncrementingCorrelationIdGenerator) _correlationIdGenerator).getCorrelationId();
+        } else {
+            throw new UnsupportedOperationException();
+        }
```

Comment:
Not super fan of this code TBH. Maybe keeping a reference to the last correlation ID (a string) and trying to convert it to an integer when `getCorrelationId()` is called would do the job as well. This is a best-effort approach anyway, but it avoids relying on some implementation.

Reply:
This is implementation is only there for BW compatiblity, So i don't think it's a problem that it only works for the 'default' id generator. I marked the method as deprecated anyway.